library/src/main/java/in/krharsh17/barview/BarView.java

```diff
@@ -56,14 +80,48 @@ void addBar(BarModel data) {
                 ViewGroup.LayoutParams.MATCH_PARENT,
                 ViewGroup.LayoutParams.WRAP_CONTENT
         ));
-
+        gestureDetector = new GestureDetector(getContext(), new SingleTapConfirm());
+
+
+        barGroup.setOnTouchListener(new OnTouchListener() {
+            private int CLICK_ACTION_THRESHOLD = 200;
+            private float startX;
+            private float startY;
+
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                switch (event.getAction()) {
+                    case MotionEvent.ACTION_DOWN:
+                        startX = event.getX();
+                        startY = event.getY();
+                        break;
+                    case MotionEvent.ACTION_UP:
+                        float endX = event.getX();
+                        float endY = event.getY();
+                        if (isAClick(startX, endX, startY, endY)) {
+                            Log.d("BarView", "you clicked!");
+                            onBarClickListener.onBarClicked(barGroups.indexOf(v));
+
+                        }
+                        break;
+                }
+                return true;
+            }
+
+            private boolean isAClick(float startX, float endX, float startY, float endY) {
+                float differenceX = Math.abs(startX - endX);
+                float differenceY = Math.abs(startY - endY);
+                return !(differenceX > CLICK_ACTION_THRESHOLD/* =5 */ || differenceY > CLICK_ACTION_THRESHOLD);
+            }
+        });
```

Comment:
Do we really need to prefer `OnTouchListener` over `OnClickListener` for what we're trying to achieve here?

Reply:
I tried using onClickListener but it was making parts of views unclickable. So I switched to OnTouchListener. If you say then I can find a workaround to replace it with `onClickListener`

Reply:
Okay, let's go with onTouchListener for now. I was only worried about beginner developers being unable to understand this.